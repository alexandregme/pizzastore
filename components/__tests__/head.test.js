import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';
import Head from '../head';

describe("Head", ()=>{
    let mountedHead;

    beforeEach(()=>{
        mountedHead = shallow(<Head />);
    });

    it('renders without crashing', () => {
        shallow(<Head />);
        expect(mountedHead).toMatchSnapshot();
    });

    it('expect to have title tag', () => {
        let title = mountedHead.find('title');
        expect(title).toHaveLength(1);
        expect(title.text()).toEqual('Pizza Store :)');
    });

    it('expect to have one charSet meta tag', () => {
        expect(mountedHead.find('meta[charSet]')).toHaveLength(1);
    });

    it('expect to have one viewport meta tag', () => {
        expect(mountedHead.find('meta[name="viewport"]')).toHaveLength(1);
    });
});

describe("Head Custom component props", ()=>{
    let mountedHead;
    let props = {
        title:'Custom title prop'
    };

    beforeEach(()=>{
        mountedHead = shallow(<Head {...props} />);
    });

    it('expect to have title tag', () => {
        let title = mountedHead.find('title');
        expect(title.text()).toEqual('Custom title prop');
    });
});