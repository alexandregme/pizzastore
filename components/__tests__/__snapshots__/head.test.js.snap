// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHead renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Head />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <title>
          Pizza Store :)
        </title>,
        <meta
          charSet="utf-8"
        />,
        <meta
          content="initial-scale=1.0, width=device-width"
          name="viewport"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pizza Store :)",
        },
        "ref": null,
        "rendered": "Pizza Store :)",
        "type": "title",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "charSet": "utf-8",
        },
        "ref": null,
        "rendered": null,
        "type": "meta",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "content": "initial-scale=1.0, width=device-width",
          "name": "viewport",
        },
        "ref": null,
        "rendered": null,
        "type": "meta",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <title>
            Pizza Store :)
          </title>,
          <meta
            charSet="utf-8"
          />,
          <meta
            content="initial-scale=1.0, width=device-width"
            name="viewport"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pizza Store :)",
          },
          "ref": null,
          "rendered": "Pizza Store :)",
          "type": "title",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "charSet": "utf-8",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "content": "initial-scale=1.0, width=device-width",
            "name": "viewport",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
